dart/collision/CollisionFilter.h

```diff
@@ -35,47 +34,35 @@
  *   POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef DART_COLLISION_COLLISIONNODE_H_
-#define DART_COLLISION_COLLISIONNODE_H_
-
-#include <cstddef>
+#ifndef DART_COLLISION_COLLISIONFILTER_H_
+#define DART_COLLISION_COLLISIONFILTER_H_
 
 namespace dart {
+
 namespace dynamics {
 class BodyNode;
 }  // namespace dynamics
-}  // namespace dart
 
-namespace dart {
 namespace collision {
 
-/// \brief
-class CollisionNode {
-public:
-  /// \brief Default constructor
-  explicit CollisionNode(dynamics::BodyNode* _bodyNode);
+class CollisionObject;
 
-  /// \brief Default destructor
-  virtual ~CollisionNode();
-
-  /// \brief
-  dynamics::BodyNode* getBodyNode() const;
-
-  /// \brief
-  void setIndex(size_t _idx);
-
-  /// \brief
-  size_t getIndex() const;
+struct CollisionFilter
+{
+  virtual bool needCollision(const CollisionObject* object1,
+                             const CollisionObject* object2) const = 0;
+};
 
-protected:
-  /// \brief
-  dynamics::BodyNode* mBodyNode;
+struct BodyNodeCollisionFilter : CollisionFilter
+{
+  bool needCollision(const CollisionObject* object1,
+                     const CollisionObject* object2) const override;
 
-  /// \brief
-  size_t mIndex;
+  bool isAdjacentBodies(const dynamics::BodyNode* bodyNode1,
+                        const dynamics::BodyNode* bodyNode2) const;
```

Comment:
Kind of a minor grammatical thing, but should we name the function ``areAdjacentBodies`` instead?